drivers/hyperv/hyperv.go

```diff
@@ -119,6 +123,14 @@ func (d *Driver) UpdateConfigRaw(rawConfig []byte) error {
 			return err
 		}
 	}
+	if newDriver.DiskCapacity != d.DiskCapacity {
+		log.Debugf("Resizing disk from %d GiB to %d GiB", d.DiskCapacity, newDriver.DiskCapacity)
```

Comment:
Should we protect against disk shrinking ?

Reply:
@guillaumerose Good point but what would be the base, like in case of vcpu and memory we fail early in crc process if user defined is less than the default value, may be in the crc codebase this is something we do for disk resize also.

Reply:
There's already a validation that the disk size is not less than the default size which is 31GB. There is no specific protection for eg resizing to 40GB and then trying to shrink to 35GB, but the libvirt machine driver will return an error in this case.